app/models/legal_aid_application.rb

```diff
@@ -397,8 +398,24 @@ def policy_disregards?
     policy_disregards&.any? ? true : false
   end
 
+  def transactions_total_by_type(transaction_type, category_id)
+    __send__("#{transaction_type}_transactions").amounts.fetch(category_id, 0)
+  end
+
   private
 
+  def bank_transactions_by_type(type)
+    bank_transactions.__send__(type).order(happened_at: :desc).by_parent_transaction_type
+  end
+
+  def cash_transactions_by_type(type)
+    cash_transactions.__send__("#{type}s").order(transaction_date: :desc).by_parent_transaction_type
+  end
+
+  def any_transactions_selected?(transaction_type, bank_trx, cash_trx)
+    !(bank_trx[transaction_type].blank? && cash_trx[transaction_type].blank?)
+  end
```

Comment:
would `bank_trx[transaction_type].any || cash_trx[transaction_type].any?` work here?  It's a bit more understandable if so.